dlt/common/normalizers/json/relational.py

```diff
@@ -252,14 +259,15 @@ def _normalize_row(
         parent_row_id: Optional[str] = None,
         pos: Optional[int] = None,
         _r_lvl: int = 0,
-        row_hash: bool = False,
+        row_id_type: TRowIdType = None,
     ) -> TNormalizedRowIterator:
         schema = self.schema
         table = schema.naming.shorten_fragments(*parent_path, *ident_path)
-        # compute row hash and set as row id
-        if row_hash:
-            row_id = self.get_row_hash(dict_row)  # type: ignore[arg-type]
-            dict_row["_dlt_id"] = row_id
+        if row_id_type in ("key_hash", "row_hash"):
```

Comment:
this logic must be moved to `_add_row_id`. I think the way we handle `_dlt_id` needs a little bit more work
- we have super simple bring your own `_dlt_id`
```
row_id = flattened_row.get("_dlt_id", None)
        if not row_id:
            row_id = self._add_row_id(table, flattened_row, parent_row_id, pos, _r_lvl)
```
we should replace it with hint based method - if there's any unique column we use it for `_dlt_id`. that may be a separate ticket. but current "bring your own" must work and now you ignore it here
- we have many methods to generate `_dlt_id` for parent table: random, deterministic (primary key based) and content based (used by scd2). they way you do it now is good enough. in the future I'd like people to be able to pick the way dlt id is generated (ie via `ItemsNormalizerConfiguration`)
- for child tables we always have deterministic `_dlt_id`

Reply:
I have moved the logic to `_add_row_id`. Current "bring your own" (if I understand it correctly) should work now.